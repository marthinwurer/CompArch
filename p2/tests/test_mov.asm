;
; tests MOV
;
; uses instructions BPT, HALT
;
; uses addressing modes 0, 2, 6
;
	.globl	start
start:	bpt
;
	mov	#1000,sp
	bpt
;
	mov	sp,r3
	bpt
;
	mov	r3,x
	bpt
;
	mov	x,r4
	bpt
;
	mov	x,y
	bpt
;
	mov	y,r5
	bpt
;
	halt
;
x:	.word	0
y:	.word	77
	.end	start
