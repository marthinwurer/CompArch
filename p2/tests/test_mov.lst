      1                                 ;
      2                                 ; tests MOV
      3                                 ;
      4                                 ; uses instructions BPT, HALT
      5                                 ;
      6                                 ; uses addressing modes 0, 2, 6
      7                                 ;
      8 000000                                   .globl  start
      9 000000  000003                  start:   bpt
     10                                 ;
     11 000002  012706  001000                   mov     #1000,sp
     12 000006  000003                           bpt
     13                                 ;
     14 000010  010603                           mov     sp,r3
     15 000012  000003                           bpt
     16                                 ;
     17 000014  010367  000030                   mov     r3,x
     18 000020  000003                           bpt
     19                                 ;
     20 000022  016704  000022                   mov     x,r4
     21 000026  000003                           bpt
     22                                 ;
     23 000030  016767  000014  000014           mov     x,y
     24 000036  000003                           bpt
     25                                 ;
     26 000040  016705  000006                   mov     y,r5
     27 000044  000003                           bpt
     28                                 ;
     29 000046  000000                           halt
     30                                 ;
     31 000050  000000                  x:       .word   0
     32 000052  000077                  y:       .word   77
     33 000054  000000'                          .end    start


pc     = 000007 %  
r0     = 000000 %  
r1     = 000001 %  
r2     = 000002 %  
r3     = 000003 %  
r4     = 000004 %  
r5     = 000005 %  
r6     = 000006 %  
r7     = 000007 %  
sp     = 000006 %  
start    000000 RG 
x        000050 R  
y        000052 R  
