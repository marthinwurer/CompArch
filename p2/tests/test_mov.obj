000 2 03 00
002 4 c6 15 00 02
006 2 03 00
008 2 83 11
00a 2 03 00
00c 4 f7 10 18 00
010 2 03 00
012 4 c4 1d 12 00
016 2 03 00
018 6 f7 1d 0c 00 0c 00
01e 2 03 00
020 4 c5 1d 06 00
024 2 03 00
026 2 00 00
028 2 00 00
02a 2 3f 00
000
