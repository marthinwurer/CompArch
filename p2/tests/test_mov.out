CPU "ARCH" Simulator, 2.5a(Jan 22 2015)
-----------------------------------------

Memory sets starting address to 0
000000:  PS[00]  000003  BPT
  R0[000000]  R1[000000]  R2[000000]  R3[000000]
  R4[000000]  R5[000000]  R6[000000]  R7[000002]
000002:  PS[00]  012706  MOV (R7)+,R6
000006:  PS[00]  000003  BPT
  R0[000000]  R1[000000]  R2[000000]  R3[000000]
  R4[000000]  R5[000000]  R6[001000]  R7[000010]
000010:  PS[00]  010603  MOV R6,R3
000012:  PS[00]  000003  BPT
  R0[000000]  R1[000000]  R2[000000]  R3[001000]
  R4[000000]  R5[000000]  R6[001000]  R7[000014]
000014:  PS[00]  010367 000030  MOV R3,000030(R7)
000020:  PS[00]  000003  BPT
  R0[000000]  R1[000000]  R2[000000]  R3[001000]
  R4[000000]  R5[000000]  R6[001000]  R7[000022]
000022:  PS[00]  016704 000022  MOV 000022(R7),R4
000026:  PS[00]  000003  BPT
  R0[000000]  R1[000000]  R2[000000]  R3[001000]
  R4[001000]  R5[000000]  R6[001000]  R7[000030]
000030:  PS[00]  016767 000014 000014  MOV 000014(R7),000014(R7)
000036:  PS[00]  000003  BPT
  R0[000000]  R1[000000]  R2[000000]  R3[001000]
  R4[001000]  R5[000000]  R6[001000]  R7[000040]
000040:  PS[00]  016705 000006  MOV 000006(R7),R5
000044:  PS[00]  000003  BPT
  R0[000000]  R1[000000]  R2[000000]  R3[001000]
  R4[001000]  R5[001000]  R6[001000]  R7[000046]
000046:  PS[00]  000000  HALT

Machine Halted - HALT instruction

Simulated time 89 cycles

LAST CPUObject DESTROYED; END OF SIMULATION
